In [31]:
import pandas as pd                   # For the dataframe structure
from functools import reduce          # For the reduce function

path = "/content/vaccinations.csv"    # The vaccination.csv dataset

df = pd.read_csv(path)                # The command used to read the csv file into the dataframe
df.head()                             # The command used to print the first 5 rows of the dataframe

location iso_code        date  total_vaccinations  people_vaccinated  \
0  Afghanistan      AFG  2021-02-22                 0.0                0.0   
1  Afghanistan      AFG  2021-02-23                 NaN                NaN   
2  Afghanistan      AFG  2021-02-24                 NaN                NaN   
3  Afghanistan      AFG  2021-02-25                 NaN                NaN   
4  Afghanistan      AFG  2021-02-26                 NaN                NaN   

   people_fully_vaccinated  total_boosters  daily_vaccinations_raw  \
0                      NaN             NaN                     NaN   
1                      NaN             NaN                     NaN   
2                      NaN             NaN                     NaN   
3                      NaN             NaN                     NaN   
4                      NaN             NaN                     NaN   

   daily_vaccinations  total_vaccinations_per_hundred  \
0                 NaN                             0.0   
1              1367.0                             NaN   
2              1367.0                             NaN   
3              1367.0                             NaN   
4              1367.0                             NaN   

   people_vaccinated_per_hundred  people_fully_vaccinated_per_hundred  \
0                            0.0                                  NaN   
1                            NaN                                  NaN   
2                            NaN                                  NaN   
3                            NaN                                  NaN   
4                            NaN                                  NaN   

   total_boosters_per_hundred  daily_vaccinations_per_million  \
0                         NaN                             NaN   
1                         NaN                            33.0   
2                         NaN                            33.0   
3                         NaN                            33.0   
4                         NaN                            33.0   

   daily_people_vaccinated  daily_people_vaccinated_per_hundred  
0                      NaN                                  NaN  
1                   1367.0                                0.003  
2                   1367.0                                0.003  
3                   1367.0                                0.003  
4                   1367.0                                0.003

In [32]:
df['date'] = df['date'].map(pd.to_datetime)   # Command used to transform the 'date' to datetime format
df.head()

location iso_code       date  total_vaccinations  people_vaccinated  \
0  Afghanistan      AFG 2021-02-22                 0.0                0.0   
1  Afghanistan      AFG 2021-02-23                 NaN                NaN   
2  Afghanistan      AFG 2021-02-24                 NaN                NaN   
3  Afghanistan      AFG 2021-02-25                 NaN                NaN   
4  Afghanistan      AFG 2021-02-26                 NaN                NaN   

   people_fully_vaccinated  total_boosters  daily_vaccinations_raw  \
0                      NaN             NaN                     NaN   
1                      NaN             NaN                     NaN   
2                      NaN             NaN                     NaN   
3                      NaN             NaN                     NaN   
4                      NaN             NaN                     NaN   

   daily_vaccinations  total_vaccinations_per_hundred  \
0                 NaN                             0.0   
1              1367.0                             NaN   
2              1367.0                             NaN   
3              1367.0                             NaN   
4              1367.0                             NaN   

   people_vaccinated_per_hundred  people_fully_vaccinated_per_hundred  \
0                            0.0                                  NaN   
1                            NaN                                  NaN   
2                            NaN                                  NaN   
3                            NaN                                  NaN   
4                            NaN                                  NaN   

   total_boosters_per_hundred  daily_vaccinations_per_million  \
0                         NaN                             NaN   
1                         NaN                            33.0   
2                         NaN                            33.0   
3                         NaN                            33.0   
4                         NaN                            33.0   

   daily_people_vaccinated  daily_people_vaccinated_per_hundred  
0                      NaN                                  NaN  
1                   1367.0                                0.003  
2                   1367.0                                0.003  
3                   1367.0                                0.003  
4                   1367.0                                0.003

In [33]:
rowsFiltered = list(filter(lambda row: row['people_fully_vaccinated_per_hundred'] > 70,     # Insert the filtered rows into a list
                           df.to_dict(orient = 'records')))                                 # The filtered rows are transformed first to dictionary so the insert values are in dictionary form
dfFiltered = pd.DataFrame(rowsFiltered)                                                     # This transforms the list into a dataframe

In [34]:
def add(x, y):        # This is a function used to add two values
  return x + y

totalVaccinated = dfFiltered['total_vaccinations'].dropna().apply(lambda x: int(x))     # Removes the NaN values and formats the numerical values to integer (non-float)
totalVaccinatedCount = reduce(add, totalVaccinated)                                     # It adds the values together of sum and new value in complete iteration
print(totalVaccinatedCount)                                                             # Prints the result of the reduce

17716869904626


# The total number of vaccinated population who has more than 70 per hundred vaccinated is **$17,716,869,904,626$**.

Functional Programming mainly focuses on creating and formulating functions that would suffice in performing tasks, especially with large datasets that would take a great deal of time if not designed to be quick or instantaneous. This approach is solely concerned with the outcome or result, whether it meets the desired or needed.

Meanwhile, Imperative Programming gives great importance on procedure, making sure that the process is meticulously followed and clear. This could be beneficial for a programming compiled to be used and debugged by non-programming professional, such as administrative and managerial roles.

However, the length needed for outlining a clear course of action could affect the runtime of the project, thus slowing down the estimated runtime of the program as compared to the Functional program.

Functional Programming is beneficial for data processing and transformation as the size of the data itself can be a factor to consider when designing a system. If the data is considered big, the processing of the said dataset could consume a large amount of time, while also factoring in the hardware used in the process. On the other hamd, if we keep the functions simple and quick, the runtime could be slimmed down to an ideal duration.